pyproject.toml

```diff
@@ -169,9 +169,15 @@ astropy_header = true
 doctest_plus = "enabled"
 text_file_format = "rst"
 remote_data_strict = true
-addopts = "--color=yes --doctest-rst"
+addopts = [
+    "-ra",
+    "--color=yes",
+    "--doctest-rst",
+    "--strict-config",
+    "--strict-markers",
+]
+log_cli_level = "info"
 xfail_strict = true
-qt_no_exception_capture = 1
```

Comment:
Old config option for `pytest-qt` (that is no-longer used) which was caught by `--strict-config`.

Reply:
We need this. Because someone installed `pytest-qt` downstream and was crashing. That's why we added this. It wasn't for us.

Reply:
A good thing to annotate in an explanatory comment!

Reply:
This option was added by #7578 in order to resolve #7572. However, IMOP `astropy` shouldn't have pytest configuration options for plugins it does not define as part of its `[test]` dependencies. If users encounter such a problem that indicates that they are mixing their astropy developer environment with other environments. This opens potentially opens the door to having lots of seemingly unrelated options configured for `pytest` for no obvious reason.

What `--strict-config` does protect us against is making things like simple typo mistakes either in the config itself or as part of CLI options. Checking for this sort of thing prevents situations where a configuration option is silently broken.

Reply:
IMO if 
https://github.com/astropy/astropy/blob/9ec4ae327bc8ca20bebbdd3987cbc14995bf1ad6/pyproject.toml#L185 is removed then this is ready to go in.